pkg/controller/statefulset/stateful_set.go

```diff
@@ -211,19 +195,18 @@ func (psc *StatefulSetController) deletePod(obj interface{}) {
 		}
 	}
 	glog.V(4).Infof("Pod %s/%s deleted through %v.", pod.Namespace, pod.Name, utilruntime.GetCaller())
-	if ps := psc.getStatefulSetForPod(pod); ps != nil {
-		psc.enqueueStatefulSet(ps)
+	if set := ssc.getStatefulSetForPod(pod); set != nil {
+		ssc.enqueueStatefulSet(set)
 	}
 }
 
 // getPodsForStatefulSets returns the pods that match the selectors of the given statefulset.
-func (psc *StatefulSetController) getPodsForStatefulSet(ps *apps.StatefulSet) ([]*v1.Pod, error) {
-	// TODO: Do we want the statefulset to fight with RCs? check parent statefulset annotation, or name prefix?
-	sel, err := metav1.LabelSelectorAsSelector(ps.Spec.Selector)
+func (ssc *StatefulSetController) getPodsForStatefulSet(set *apps.StatefulSet) ([]*v1.Pod, error) {
+	sel, err := metav1.LabelSelectorAsSelector(set.Spec.Selector)
 	if err != nil {
 		return []*v1.Pod{}, err
 	}
-	pods, err := psc.podStore.Pods(ps.Namespace).List(sel)
+	pods, err := ssc.podStore.Pods(set.Namespace).List(sel)
 	if err != nil {
 		return []*v1.Pod{}, err
 	}
```

Comment:
The copy below is 1. shallow and 2. is probably not needed (I saw you are deep-copying in the ss controller interface)